pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -220,3 +230,91 @@ func (r *ReconcileKogitoApp) setDefaultBuildLimits(instance *v1alpha1.KogitoApp)
 		}
 	}
 }
+
+// Ensure that all Kogito images are available before the build.
+func (r *ReconcileKogitoApp) ensureKogitoImageStream(instance *v1alpha1.KogitoApp) (reconcile.Result, error) {
+	need2createIS := false
+
+	imageStreamTag := kogitores.ImageStreamTag
+	if instance.Spec.Build.ImageS2I.ImageStreamTag != "" {
+		imageStreamTag = instance.Spec.Build.ImageS2I.ImageStreamTag
+		log.Infof("Using ImageS2I Tag from CR: %s", imageStreamTag)
+	}
+
+	// look for the Kogito image stream on the default namespace (openshift)
+	// All images must be present, if not create it on current namespace
+	for _, key := range kogitores.ImageStreamNameList {
+		isOnOCPNamespace, _ := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: key, Namespace: kogitores.ImageStreamNamespace}, imageStreamTag)
+		if isOnOCPNamespace == nil {
+			need2createIS = true
+		}
+	}
+
+	if need2createIS {
+		imageStreamLocation := fmt.Sprintf(kogitores.ImageStreamLocation, imageStreamTag)
+		log.Infof("Using imagestream %s", imageStreamLocation)
+		resp, err := http.Get(imageStreamLocation)
+		if err != nil {
+			log.Errorf("Failed to fetch imagestream %s", err.Error())
+			return reconcile.Result{}, err
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode == http.StatusOK {
+
+			var imagestreamlist oimagev1.ImageStreamList
+			bodyBytes, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				log.Warnf("Failed to read response: %s", err.Error())
```

Comment:
Wouldn't it have sense to return from method in case response body cannot be read? In this implementation only warning is shown and method execution continue with decoding of body content.

Reply:
+1, updated.